src/heros/solver/IDESolver.java

```diff
@@ -746,10 +746,14 @@ private void propagateValueAtCall(Pair<N, D> nAndD, N n) {
 		}
 	}
 	
+	protected V joinValueAt(N unit, D fact, V curr, V newVal) {
+		return valueLattice.join(curr, newVal);
+	}
+	
 	private void propagateValue(N nHashN, D nHashD, V v) {
 		synchronized (val) {
 			V valNHash = val(nHashN, nHashD);
-			V vPrime = valueLattice.join(valNHash,v);
+			V vPrime = joinValueAt(nHashN, nHashD, valNHash,v);
```

Comment:
One could imagine other places where it makes sense to hook into the value computation phase. However, this seems the most natural starting point.